enterprise/frontend/src/metabase-enterprise/whitelabel/lib/whitelabel.js

```diff
@@ -2,7 +2,7 @@ import { colors } from "metabase/lib/colors/palette";
 import MetabaseSettings from "metabase/lib/settings";
 
 export function updateColors() {
-  const scheme = MetabaseSettings.get("application-colors");
+  const scheme = MetabaseSettings.get("application-colors") || {};
```

Comment:
Just a curiosity question, but I think this field is unset when the application colors haven't been overridden in the appearance settings? So, I think this value is only populated when the users set them in the admin panel? If so, using a default like this should work fine.

cc @WiNloSt we were discussing about this issue in Slack/GitHub earlier

Reply:
Yeah, just retested. And also from the getter, it will always return a map.
https://github.com/metabase/metabase/blob/44cb385c40d316b6fbc1986caa00cf9ba8d0a04d/src/metabase/public_settings.clj#L341-L349

Reply:
@WiNloSt it was returning `undefined` or `null` in the SDK though, so I think that is why Denis added the `|| {}` part.

Reply:
adding `|| {}` looks like a hack to me if we expect `MetabaseSettings.get('application-colors ')` to always return a map.

By doing this wouldn't we just ignore the fact that `application-colors` doesn't return the correct value? That means when users set colors in appearance settings, if we don't provide `theme` would those colors even work?

Reply:
That's what I wonder as well. My guess is that if `application-colors` is not set, then the default colors defined in `colors` would get applied. However, when one or more colors is set then it's going to return a map. That would explain this diff I saw above as well. Would be great if @deniskaber could confirm our assumptions here 🙏🏻 

```ts
    } else {
      Object.entries(originalColors).forEach(([key, value]) => {
        colors[key as keyof MetabaseColors] = value;
      });
    }
```

Reply:
I can confirm custom colors doesn't work at the moment. Just to be clear this is the case when users set custom colors on the instance and expect those colors to propagate to the SDK, it won't work.

#### SDK
![image](https://github.com/metabase/metabase/assets/1937582/25929d03-0547-43c2-a45d-88ee0af7c930)

#### App
![image](https://github.com/metabase/metabase/assets/1937582/21502b84-50ef-4c56-a92b-0ea6c98a01ca)

#### Color settings
![image](https://github.com/metabase/metabase/assets/1937582/36c54f60-851b-4933-9fdd-6d3abb8e03e1)

So, I think we should get to the root cause why this doesn't work rather than adding `|| {}` which like just to silent the error, but still doesn't set the correct values.

Reply:
Good catch @WiNloSt! This was on my list of things to tackle after I finished customer zero deployment, but I thought this PR fixes it. My bad. Let me update https://github.com/metabase/metabase/issues/42705 to reflect this.

Reply:
Sorry, I was testing with `master` I'll test it with this branch and I'll let you know the result.

Reply:
Well I don't think the color from instance is working 
![image](https://github.com/metabase/metabase/assets/1937582/a6fb83cd-2e00-4e72-ab9a-b329a4cb14be)

Also, this seems to make the admin settings look off. Can you confirm that too @heypoom ?
![image](https://github.com/metabase/metabase/assets/1937582/09458370-7f6a-437f-a1a4-3367e890da6e)

Reply:
@WiNloSt yes, can confirm on my end that this breaks the dashboard. Thanks for running this locally!

![CleanShot 2567-05-16 at 17 59 55@2x](https://github.com/metabase/metabase/assets/4714175/d796329a-ce2d-4255-9daa-8aec91afd493)

![CleanShot 2567-05-16 at 18 00 12@2x](https://github.com/metabase/metabase/assets/4714175/d7bd08da-fcd4-42b0-9e34-e58289cf3dda)

Reply:
@deniskaber I see, thanks for the explanation! As I will be working on #42705 next, do you know if there's a we can retrieve the `application-colors` object (i.e. `{{bootstrapJSON}}`) dynamically rather than relying on `index_template.html`? So I can update the `updateColors` logic accordingly.

I think it's OK if this PR doesn't cover the customized ui colors logic, I can tackle that in my PR later. I think we can stick with `|| {}` for now just to not throw the error, no need to revert it. 

We can focus on resolving the dashboard styling issue in the above screenshot for now 🙏🏻